src/lib.rs

```diff
@@ -516,9 +516,15 @@ impl<TypeState> ConnectionOptions<TypeState> {
         self
     }
 
-    /// Set a callback to be executed when the client has been
-    /// closed due to exhausting reconnect retries to known servers
-    /// or by completing a drain request.
+    /// Set a callback to be executed for calculating the backoff duration
+    /// to wait before a server reconnection attempt.
```

Comment:
cluster, since it makes it clear that this callback should fire only after all servers were not able to connect.